bundles/org.smarthomej.binding.tuya/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -185,4 +185,30 @@
 		</config-description>
 	</channel-type>
 
+	<channel-type id="ircode1">
+		<item-type>String</item-type>
+		<label>IR Code (Template)</label>
+		<description>Code library parameter</description>
+
+		<config-description>
+			<parameter name="irCode" type="text" required="true">
+				<label>IR Code</label>
+				<description>Decoding parameter</description>
+			</parameter>
+			<parameter name="dp" type="integer" required="true">
```

Comment:
Why do you name these `dp` and `dp2`? I don't see an immediate need for that.

Reply:
to re-use existing parameters

Reply:
I don't think that's necessary. Just name it meaningful. If you encounter a problem, just ask.

Reply:
Can you adjust that, so we can merge here? I would like to do a new release this weeks it would be great to have it in.

Reply:
Hi, added a new PR https://github.com/smarthomej/addons/pull/499